docker/dev/docker-compose.yml

```diff
@@ -1,75 +1,109 @@
-version: '3.3'
+version: "3.3"
+
+networks:
+  # communication to web and clients
+  lemmyexternalproxy:
+  # communication between lemmy services
+  lemmyinternal:
+    driver: bridge
+    internal: true
 
 services:
-  nginx:
+  lemmy-proxy:
     image: nginx:1-alpine
+    container_name: lemmy-proxy
+    networks:
+      - lemmyinternal
+      - lemmyexternalproxy
     ports:
-      - "1236:1236"
+    # actual and only port facing any connection from outside 
+      - "1239:1236"
     volumes:
-      - ./nginx.conf:/etc/nginx/nginx.conf
-    restart: always
+      - ./nginx.conf:/etc/nginx/nginx.conf:ro
+      - ./nginx_proxy.conf:/etc/nginx/proxy.conf:ro
+    restart: unless-stopped
     depends_on:
-      - pictrs
+      - lemmy-pictrs
       - lemmy-ui
 
   lemmy:
-    image: lemmy-dev:latest
-    ports:
-      - "8536:8536"
-      - "6669:6669"
-    restart: always
+    image: dessalines/lemmy:dev
+    container_name: lemmy
+    hostname: lemmy
+    networks:
+      - lemmyinternal
+    restart: unless-stopped
```

Comment:
In my experience there is no practical difference between always and unless-stopped, but maybe this is clearer.

Reply:
_Similar to always, except that when the container is stopped (manually or otherwise), it is not restarted even after Docker daemon restarts._

It's not super relevant in this scenario i agree. Maybe for jaeger and otel - if someone does not need/want these they would not start again after each daemon restart (+ reboot).

Reply:
Ah i didnt know that.